src/core/payoutCalculator/PayoutCalculatorIsolateSuperCharge.ts

```diff
@@ -19,22 +19,32 @@ export class PayoutCalculatorIsolateSuperCharge implements IPayoutCalculator {
         o1CommissionRate: number,
         commissionRates: KeyCommissionRate,
     ): Promise<
-        [payoutJson: PayoutTransaction[], storePayout: PayoutDetails[], blocksIncluded: number[], totalPayout: number]
+        [payoutJson: PayoutTransaction[], storePayout: PayoutDetails[], blocksIncluded: number[], totalPayout: number, totalToBurn: number]
     > {
         //TODO: JC - Shared Logic must be moved into its own class, then isolate change in behaviors
         // Initialize some stuff
         const blocksIncluded: number[] = [];
         const storePayout: PayoutDetails[] = [];
+        let sumCoinbase = 0;
+        let sumCoinbaseNoSuperchargedRewards = 0;
 
         // for each block, calculate the effective stake of each staker
         blocks.forEach((block: Block) => {
             // Keep a log of all blocks we processed
             blocksIncluded.push(block.blockheight);
-
+            
             if (typeof block.coinbase === 'undefined' || block.coinbase == 0) {
                 // no coinbase, don't need to do anything
             } else {
                 const winner = this.getWinner(stakers, block);
+                const coinbase = block.coinbase;
+                const winnerStakeIsLocked = stakeIsLocked(winner, block);
+                const winnerIsMf = winner.shareClass.shareOwner == 'MF' ? true : false;
+
+                const burnWinnerScr = (commissionRates[winner.publicKey]) ? commissionRates[winner.publicKey].burnSuperchargedRewards : 0;
+                const burnScr = (coinbase > 720000000000) && (!winnerStakeIsLocked) && (winnerIsMf || burnWinnerScr);
```

Comment:
>720, or == 1440? I think = 1440 would be a better guard since that is what we expect and anything else may mean this behavior will be unintended/unexpected.

Reply:
Done